cpp/include/cudf/scalar/scalar.hpp

```diff
@@ -395,81 +395,96 @@ class string_scalar : public scalar {
    * @param[in] stream CUDA stream used for device memory operations.
-   * @param[in] mr Device memory resource to use for device memory allocation
+   * @param[in] mr Device memory resource to use for device memory allocation.
    */
   string_scalar(std::string const& string,
                 bool is_valid                       = true,
                 rmm::cuda_stream_view stream        = rmm::cuda_stream_default,
                 rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
   /**
-   * @brief Construct a new string scalar object from string_view
-   * Note that this function copies the data pointed by string_view.
+   * @brief Construct a new string scalar object from string_view. Note that this function copies
+   * the data pointed by string_view.
    *
-   * @param[in] source string_view pointing string value to copy
-   * @param[in] is_valid Whether the value held by the scalar is valid
+   * @param[in] source The string_view pointing the string value to copy.
+   * @param[in] is_valid Whether the value held by the scalar is valid.
    * @param[in] stream CUDA stream used for device memory operations.
-   * @param[in] mr Device memory resource to use for device memory allocation
+   * @param[in] mr Device memory resource to use for device memory allocation.
    */
   string_scalar(value_type const& source,
                 bool is_valid                       = true,
                 rmm::cuda_stream_view stream        = rmm::cuda_stream_default,
                 rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
   /**
-   * @brief Construct a new string scalar object from string_view in device memory
-   * Note that this function copies the data pointed by string_view.
+   * @brief Construct a new string scalar object from string_view in device memory. Note that this
+   * function copies the data pointed by string_view.
    *
-   * @param[in] data device_scalar string_view pointing string value to copy
-   * @param[in] is_valid Whether the value held by the scalar is valid
+   * @param[in] data The device_scalar of string_view pointing to the string value to copy.
+   * @param[in] is_valid Whether the value held by the scalar is valid.
    * @param[in] stream CUDA stream used for device memory operations.
-   * @param[in] mr Device memory resource to use for device memory allocation
+   * @param[in] mr Device memory resource to use for device memory allocation.
    */
   string_scalar(rmm::device_scalar<value_type>& data,
                 bool is_valid                       = true,
                 rmm::cuda_stream_view stream        = rmm::cuda_stream_default,
                 rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
   /**
-   * @brief Implicit conversion operator to get the value of the scalar in a host std::string
+   * @brief Construct a new string scalar object from an orphaned buffer in device memory.
```

Comment:
Let's not add potentially confusing terminology ("orphaned"). 
```suggestion
   * @brief Construct a new string scalar object by moving an existing string data buffer.
```